llvm/lib/Transforms/Yk/ControlPoint.cpp

```diff
@@ -317,6 +327,15 @@ class YkControlPoint : public ModulePass {
     IRBuilder<> Builder(Caller->getEntryBlock().getFirstNonPHI());
     Value *InputStruct = Builder.CreateAlloca(CtrlPointVarsTy, 0, "");
 
+    // Also at the top, generate storage for the interpreted return of the control points caller.
+    Value *ReturnPtr;
```

Comment:
Can't this be just `Value *ReturnPtr = Builder.CreateAlloca(ReturnTy, 0, "")` since the `void` case has been handled above?

Reply:
The above handles the signature of the control point call. This handles the alloca in which we store the interpreted return value. I can probably move the below up and reuse the `if` block.

Reply:
Not sure I understand.

Reply:
Not sure how to explain. The short answer is: No it can't. ;-)

The longer answer is that we can't just do `Builder.CreateAlloca(ReturnTy, 0, "")` if `ReturnType = void`, since you can't allocate `void`. So we need the extra check here.

Reply:
But haven't we converted `void` return to `int` return earlier?

Reply:
We have, but I haven't stored it in a variable to make things less confusing. Okay, let me try something else and see if that makes things better.

Reply:
Try this a55604